app/models/miq_queue_worker_base/runner.rb

```diff
@@ -110,7 +110,6 @@ def deliver_queue_message(msg, &block)
         end
       end
 
-      msg.delivered(status, message, result) unless status == MiqQueue::STATUS_RETRY
```

Comment:
This is a slight change in behvior that I'm not sure is important or not.

Before, if a timeout occurred:
1. `.deliver` happens and returns status timeout
2. timeout is logged
3. connection is reset and spid is updated (potentially exiting the process)
4. message is marked as `.delivered` (state is set, callback is called, stale message is deleted).

After. if a timeout occurred:
1. `.deliver` happens and returns status timeout
2. message is marked as `.delivered` (state is set, callback is called, stale message is deleted).
3. timeout is logged
4. connection is reset and spid is updated (potentially exiting the process)

I'm not sure if that will be a problem or not

Reply:
I think there is something here.
Probably could have handled this. (e.g.: off loading `delivered` on the caller when a `timeout` occurs)

Moving this to just tests.